File: February 2023/Day 08/Count number of free cell/count-number-of-free-cell.java
//{ Driver Code Starts
//Initial Template for Java


import java.io.*;
import java.util.*;

public class GFG {
    public static void main(String[] args) throws Exception {
        BufferedReader in=new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out=new PrintWriter(System.out);
        int T = Integer.parseInt(in.readLine().trim());
        while(T>0)
        {
            String s[]=in.readLine().trim().split(" ");
            int n = Integer.parseInt(s[0]);
            int k = Integer.parseInt(s[1]);
            int[][] a = new int[k][2];
            for(int i=0; i<k; i++){
                s=in.readLine().trim().split(" ");
                a[i][0] = Integer.parseInt(s[0]);
                a[i][1] = Integer.parseInt(s[1]);
            }
            Solution g = new Solution();
            long[] res = g.countZero(n,k,a);
            
            for(int i=0; i<res.length; i++){
                out.print(res[i]+" ");
            }
            out.println();
            T--;
        }
        out.close();
    }
}
// } Driver Code Ends


//User function Template for Java


class Solution {
    long[] countZero(int N, int K, int[][] arr){
        // code here
        HashSet<Integer> row = new HashSet<>();
        HashSet<Integer> col = new HashSet<>();
        long[] ans = new long[K];
        long n =N;
        long total_zero=n*n;
        long total_one,r,c;
        for(int i=0;i<K;i++) {
            row.add(arr[i][0]);
            col.add(arr[i][1]);
            r=row.size();
            c=col.size();
            total_one = ((r+c)*n) - (r*c);
            ans[i]=total_zero - total_one;
        }
        return ans;
    }
}

File: February 2023/Day 08/Count number of free cell/count-number-of-free-cell.cpp
//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h>
using namespace std;



// } Driver Code Ends
//User function Template for C++


class Solution{
  #define ll long long int
  public:
  vector<long long int> countZero(int N, int K, vector<vector<int>>& arr){
      //Code Here
    unordered_set<int> row;
    unordered_set<int> col;
    vector<ll> ans(K,0);
    ll n =N;
    ll total_zero=n*n;
    ll total_one,r,c;
    for(int i=0;i<K;i++) {
        row.insert(arr[i][0]);
        col.insert(arr[i][1]);
        r=row.size();
        c=col.size();
        total_one = ((r+c)*n) - (r*c);
        ans[i]=total_zero - total_one;
    }
    return ans;
  }
};

//{ Driver Code Starts.
int main(){
    int t;
    cin>>t;
    while(t--){
        int n,k;
        cin>>n>>k;
        vector<vector<int>> arr(k,vector<int>(2));
        for(int i=0;i<k;i++){
            int x, y;
            cin>>x>>y;
            arr[i][0] = x;
            arr[i][1] = y;
        }
        Solution ob;
        vector<long long int> ans = ob.countZero(n,k,arr);
        
        for(int i=0;i<ans.size();i++){
            cout<<ans[i]<<" ";
        }
        cout<<endl;    
    }
    
    return 0;
}


// } Driver Code Ends